src/parsers/parseOneOf.ts

```diff
@@ -20,12 +20,13 @@ export const parseOneOf = (
         }),
       )
       .join(", ")}];
-    const errors = schemas.reduce(
-      (errors: z.ZodError[], schema) =>
-        ((result) => ("error" in result ? [...errors, result.error] : errors))(
-          schema.safeParse(x)
+    const errors = schemas.reduce<z.ZodError[]>(
+      (errors, schema) =>
+        ((result) =>
+          "error" in result && typeof result.error !== 'undefined' ? [...errors, result.error] : errors)(
```

Comment:
i'm not sure why you need so many hacks in order to tell TS that property exists
```suggestion
          result.error ? [...errors, result.error] : errors)(
```
will do the trick, because:
problem is that this type
`export declare type SafeParseReturnType<Input, Output> = SafeParseSuccess<Output> | SafeParseError<Input>;`

from zod returns one of 2 structures:
```
type SafeParseSuccess<Output> = {
    success: true;
    data: Output;
    error?: undefined;
}
```

```
type SafeParseError<Input> = {
    success: false;
    error: ZodError<Input>;
    data?: undefined;
}
```

the only thing we need is to tell typescript "we're using SafeParseError here"

TS is smart enough to figure out diff between types and use correct one

so in this case just checking properly (instead of ancient "in" syntax) will do the trick

Reply:
You're right that is simpler, I'll make the change.